src/transformers/visitors/toContractVisitor.ts

```diff
@@ -739,13 +737,28 @@ export default {
           ]
           newNode = buildNode('InternalFunctionCall', {
           name: node.expression.name,
-          internalFunctionInteractsWithSecret: internalFuncInteractsWithSecret,
+          internalFunctionInteractsWithSecret: state.internalFunctionInteractsWithSecret,
           parameters: state.fnParameters,
          });
          node._newASTPointer = newNode;
          parentnewASTPointer(parent, path, newNode , parent._newASTPointer[path.containerName]);
          return;
-        }
+       } else if(!state.internalFunctionInteractsWithSecret){
+         state.internalFncName ??= [];
+         state.internalFncName.push(node.expression.name);
+         const fnDefNode = path.getAncestorOfType('FunctionDefinition');
+         state.callingFncName ??= [];
+         state.callingFncName.push(fnDefNode.node.name);
+         newNode = buildNode('InternalFunctionCall', {
```

Comment:
Could you check for non secret internalFunctionCall that contain parameters? may need to modify codegeneration since parameters list are stored differently .